### PR TITLE
[Build] Add cmake options into libinfo

### PR DESCRIPTION
**Pull request by junrushao1994**
To follow up with #6280, I added cmake options into libinfo, then if we execute the following command, we will have the full build info:

```
> python -c "import tvm; print('\n'.join(k + '\t' + v for k, v in tvm.support.libinfo().items()))"
USE_LLVM	llvm-config --ignore-libllvm
GIT_COMMIT_HASH	f812dacc024812ea02822fb8f447bf5a8c607f2c
USE_VULKAN	OFF
USE_HEXAGON_SDK	/path/to/sdk
USE_MICRO	OFF
USE_GRAPH_RUNTIME	ON
INSTALL_DEV	OFF
USE_SORT	ON
USE_OPENCL	OFF
USE_THREADS	ON
USE_CUDNN	OFF
USE_RANDOM	OFF
USE_STACKVM_RUNTIME	OFF
USE_GRAPH_RUNTIME_DEBUG	ON
USE_THRUST	OFF
USE_MICRO_STANDALONE_RUNTIME	OFF
USE_TENSORFLOW_PATH	none
USE_MSVC_MT	OFF
USE_CUDA	OFF
USE_TF_TVMDSOOP	OFF
USE_COREML	OFF
USE_HEXAGON_DEVICE	OFF
USE_MIOPEN	OFF
USE_DNNL_CODEGEN	OFF
USE_ROCM	OFF
USE_BLAS	none
USE_TFLITE	OFF
USE_RPC	ON
USE_ARM_COMPUTE_LIB_GRAPH_RUNTIME	OFF
USE_RELAY_DEBUG	ON
USE_TARGET_ONNX	OFF
USE_ROCBLAS	OFF
USE_MKLDNN	OFF
USE_ARM_COMPUTE_LIB	OFF
USE_FALLBACK_STL_MAP	OFF
USE_CUBLAS	OFF
USE_RTTI	ON
USE_OPENMP	none
USE_MKL	OFF
USE_METAL	OFF
HIDE_PRIVATE_SYMBOLS	OFF
USE_NNPACK	OFF
ROCM_PATH	/opt/rocm
USE_CPP_RPC	OFF
```

CC: @jroesch @jroesch @tqchen

### CODE REVIEW

File: src/support/libinfo.cc [hunk 1]
```diff
@@ -24,11 +24,230 @@
 #define TVM_GIT_COMMIT_HASH "NOT-FOUND"
 #endif
 
+#ifndef TVM_USE_CUDA
+#define TVM_USE_CUDA "NOT-FOUND"
+#endif
+
+#ifndef TVM_USE_OPENCL
```
Comment (tqchen) [inline src/support/libinfo.cc:31]: Let us change the macro prefix, as TVM_USE_XYZ is used for normal command switch. Instead, let us use TVM_INFO_USE_OPENCL

Comment (junrushao1994) [reply src/support/libinfo.cc]: It makes sense. Fixed. Thank you!

**Comment by comaniac** [comment]
This may be out of scope. For example, USE_LLVM=llvm-config provides the command (or a path) to build with LLVM, but LLVM=9.0 is way more useful. Since now every flag is explicitly specified in libinfo, is that possible to customize the displayed information of some flags?

**Comment by junrushao1994** [comment]
Hey @comaniac, I think you proposed two improvements
1) Incorporate LLVM version: it is easy to do by adding `TVM_LLVM_VERSION` to libinfo. It will be added in this PR.
2) Potential ways to customize the information in libinfo: It is a broader question that I may have no perfect answer, because those flags appears in very diverse ways in cmake...But it doesn't mean it is hard to manually do whatever you want if you slightly modify cmake scripts. For example, you may just add a single line in `LibInfo.cmake` and another line in `libinfo.cc` to add a flag.

**Comment by comaniac** [comment]
Hey thanks for the change. I agree with your methodology that we could customize the libinfo to deliver whatever we want. Apparently, it's impractical to parse camel.config and we have to manually setup the libinfo. With this statement, we should be able to have more flexibility in libinfo. For example, we even don't need USE_LLVM, USE_CUDA corresponding to cmake flags, but just need LLVM_VERSION, CUDA_VERSION.

I think it's worthwhile to have an RFC discussing the libinfo contents (or we could refer to a previous RFC about benchmarking guidelines). However, this PR is a good start and shouldn't be blocked by this topic.

**Comment by junrushao1994** [comment]
@comaniac I agree. It would be helpful if we have a forum thread discussing what should be included in the libinfo. A potentially issue is that I might not have all the devices required to verify correctness, for example, I don't have an ARM device at hand...Therefore, help from you guys and the community would be much appreciated!

**Pull request merged**